spv/sync.go

```diff
@@ -480,7 +480,9 @@ var errBreaksMinVersionTarget = errors.New("peer uses too low version to satisif
 
 // connectAndRunPeer connects to and runs the syncing process with the specified
 // peer. It blocks until the peer disconnects and logs any errors.
-func (s *Syncer) connectAndRunPeer(ctx context.Context, raddr string, persistent bool) {
+// connectAndRunPeer returns backoff flag indicating whether it is advised to
+// wait a bit before attempting to connect to this peer again.
+func (s *Syncer) connectAndRunPeer(ctx context.Context, raddr string, persistent bool) (backoff bool) {
```

Comment:
Essentially, `backoff` flag will help the caller to decide whether to retry immediately or wait (seems like there are cases for both).